test/tests/dash_c.py

```diff
@@ -0,0 +1,18 @@
+import sys
+import subprocess
+
+me = sys.executable
+
+with open('/dev/null')as ignore:
+    # We don't (yet?) require exact stderr or return code compatibility w/
+    # python. So we just check that we succeed or fail as appropriate.
+    def do(args):
+        print 0 == subprocess.call([me] + args, stderr=ignore)
+
+    run(["-c", "print 2 + 2"])
```

Comment:
I assume you meant `do()` (or to call the function `run()`).  Maybe we could add a tester annotation that says that a script is allowed to have a nonzero exit code?  Currently things will pass if CPython and Pyston both fail, which they do here but is unexpected.